### PR TITLE
Published docs and updated repository structure to new template

### LINKED ISSUE #79: [Docs]: setup readthedocs
### Version checks

- [x] I have checked that the issue still exists on the latest versions on read the docs.

### Location of the documentation

Documentation isn't available online yet.

### Documentation problem

Doesn't exist yet.

### Suggested fix for documentation

Setup the documentation folders following the [cookiecutter-caf template](https://github.com/Transport-for-the-North/cookiecutter-caf/) and setup the `.readthedocs.yaml` so that the docs can be generated on readthedocs, see [caf.toolkit `.readthedocs.yaml` for example](https://github.com/Transport-for-the-North/caf.toolkit/blob/main/.readthedocs.yaml).

### LINKED ISSUE #90: [Feat]: update to new TfN Python template
### Feature Type

- [x] Adding new functionality

- [x] Changing existing functionality

- [ ] Removing existing functionality

### Problem Description

CAF.base was setup with the old TfN Python repository template.

### Feature Description

Update repository structure to match the new [cookiecutter-caf template](https://github.com/Transport-for-the-North/cookiecutter-caf/), updates include:
- New documentation structure
- Python examples folder and inclusion in documentation
- Dynamic dependencies in pyproject.toml
- Updated README
- Centralised GitHub issues / PR templates
- Removal of `setup.cfg` and `tox.ini`

Can use cookiecutter to automate the updates, although some manual changes will be required to integrate with the exists files.

Run the template with: `cookiecutter gh:Transport-for-the-North/cookiecutter-caf` 

### Alternative Solutions

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by MattBuckley-TfN**
Describe Changes
---
Ran cookiecutter to update template and add generic docs structure.

- Closes #90 
- Closes #79 

[**Published branch docs**](https://cafbase--114.org.readthedocs.build/en/114/index.html)

Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Have integration tests been added (if applicable, to test modules of functionality)
- [ ] Updated RELEASE.md to reflect changes in PR
- [ ] Have new dependencies been added?
  - [ ] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [ ] Have they been added to `setup.cfg`

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+dir_path = pathlib.Path(__file__).parents[2]
+source = dir_path / "src"
+sys.path.insert(0, str(source.absolute()))
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = "CAF.base"
+copyright = "2024, Transport for the North"
```
Comment (BenTaylor-TfN) [inline docs/source/conf.py]:
```suggestion
copyright = "2025, Transport for the North"
```

**Review by BenTaylor-TfN — APPROVED** [review]
